### PR TITLE
Refactor/use format

### LINKED ISSUE #1081: Use `std::format`
We have a bunch of warnings in [the sonarcloud report](https://sonarcloud.io/project/issues?resolved=false&rules=cpp%3AS6185&id=VROOM-Project_vroom) about using `std::format`. I did not address this earlier in #984 or #986 simply because the support was missing in compilers at that time.

I suspect newer versions of both gcc and clang would now be OK with the changes so we should dig into this once the CI compiler upgrade in #1080 is completed.

### PR DESCRIPTION
**Pull request by SebMilardo**
## Issue

Fixes #1081 

## Tasks

 - [x] Use std::format instead of concatenating pieces manually
 - [x] review

### CODE REVIEW

File: src/structures/vroom/input/input.cpp [hunk 1]
```diff
@@ -186,47 +188,51 @@ void Input::add_job(const Job& job) {
...
+      std::format("Inconsistent shipment amount for pickup {} and delivery {}.",
+                  pickup.id,
+                  delivery.id));
   }
   if (pickup.skills.size() != delivery.skills.size()) {
-    throw InputException("Inconsistent shipment skills for pickup " +
-                         std::to_string(pickup.id) + " and delivery " +
-                         std::to_string(delivery.id) + ".");
+    throw InputException(
+      std::format("Inconsistent shipment skills for for pickup {} and delivery "
```
Comment (jcoupey) [inline src/structures/vroom/input/input.cpp:205]: Typo here with extra "for".

**Review by jcoupey — COMMENTED** [review]
Thanks for the PR! Looks great overall, the error messages are much more readable now.

Sonarcloud [is complaining](https://sonarcloud.io/project/issues?id=VROOM-Project_vroom&pullRequest=1084&resolved=false&inNewCodePeriod=true) about not using raw string literals to avoid escaping, but I don't know how this plays along with `std::format`?

**Comment by SebMilardo** [comment]
> Thanks for the PR! Looks great overall, the error messages are much more readable now.
> 
> Sonarcloud [is complaining](https://sonarcloud.io/project/issues?id=VROOM-Project_vroom&pullRequest=1084&resolved=false&inNewCodePeriod=true) about not using raw string literals to avoid escaping, but I don't know how this plays along with `std::format`?

It looks like I have to rewrite them as:

`R"({{"lon":{},"lat":{},"type":"break"}},)"`

and

`R"({{"lon":{},"lat":{}}},)"`

Let me push a commit...

**Comment by jcoupey** [comment]
Thanks for the changes, looks all good to me.

The only weird thing is we'll have to wait to merge since our compiler versions in CI will error on `std::format` currently. Thus there is no point in enabling the CI checks right now either since they're bound to fail. Let's leave this open for now and revisit once #1080 lands.

**Comment by jcoupey** [comment]
We should now be able to let this run through the CI checks after upgrading compiler versions in #1113.

@SebMilardo could you pull the latest master on your fork and `git merge --no-ff master` into your PR branch? I did just that but it looks like I'm not allowed to push to your fork.

**Comment by SebMilardo** [comment]
It should be ok now

File: src/routing/valhalla_wrapper.cpp [hunk 2]
```diff
@@ -37,8 +37,8 @@ std::string ValhallaWrapper::get_matrix_query(
   // List locations.
   std::string all_locations;
   for (auto const& location : locations) {
-    all_locations += "{\"lon\":" + std::to_string(location.lon()) + "," +
-                     "\"lat\":" + std::to_string(location.lat()) + "},";
+    all_locations +=
+      std::format(R"({{"lon":{},"lat":{}}},)", location.lon(), location.lat());
```
Comment (jcoupey) [inline src/routing/valhalla_wrapper.cpp:41]: Did you check this produces the same behavior as before? Asking before I don't really get the reason we have to use `{{` and `}}`.

**Review by jcoupey — COMMENTED** [review]
Thanks for the update, CI builds are OK now! \o/

**Review by jcoupey — APPROVED** [review]

**Pull request merged**